### PR TITLE
SIGNL4 added as Solution

### PR DESCRIPTION
**Pull request by rons4**
SIGNL4 added as solution.

## Proposed Changes

This adds the mobile alerting and incident response service SIGNL4 as a solution .

### CODE REVIEW

File: Solutions/SIGNL4/Data Connectors/Logo/signl4.svg [hunk 1]
```diff
@@ -0,0 +1,224 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (NikTripathi) [inline Solutions/SIGNL4/Data Connectors/Logo/signl4.svg:1]: Please reduce the size of logo. Logo should be less than 5kb. Please refer [guidelines](https://microsoft.sharepoint.com/:x:/r/teams/AzureSecurityInsights/_layouts/15/guestaccess.aspx?e=aETsaQ&share=EWs_HuWia89Ns0pc1jXRWQkB7aNXX6Zi1REQcKtH4gQp5Q&isSPOFile=1)

File: Solutions/SIGNL4/Data Connectors/Logo/signl4.svg [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+     fit-margin-left="0"
+     fit-margin-right="0"
+     fit-margin-bottom="0"
+     inkscape:window-width="1286"
+     inkscape:window-height="786"
+     inkscape:window-x="182"
+     inkscape:window-y="40"
+     inkscape:window-maximized="0" />
+  <metadata
+     id="metadata5">
```
Comment (NikTripathi) [inline Solutions/SIGNL4/Data Connectors/Logo/signl4.svg:44]: Please update all id in a guid format. Thanks

File: Solutions/SIGNL4/Data Connectors/Logo/signl4.svg [hunk 3]
```diff
@@ -0,0 +1,224 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!-- Created with Inkscape (http://www.inkscape.org/) -->
+
+<svg
+   xmlns:dc="http://purl.org/dc/elements/1.1/"
+   xmlns:cc="http://creativecommons.org/ns#"
+   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
+   xmlns:svg="http://www.w3.org/2000/svg"
+   xmlns="http://www.w3.org/2000/svg"
```
Comment (NikTripathi) [inline Solutions/SIGNL4/Data Connectors/Logo/signl4.svg:9]: Please remove xmlns link. Please refer [guidelines](https://microsoft.sharepoint.com/:x:/r/teams/AzureSecurityInsights/_layouts/15/guestaccess.aspx?e=aETsaQ&share=EWs_HuWia89Ns0pc1jXRWQkB7aNXX6Zi1REQcKtH4gQp5Q&isSPOFile=1).

**Review by NikTripathi — CHANGES_REQUESTED** [review]

**Comment by NikTripathi** [comment]
@anki-narravula Can you please review playbooks. Thanks

File: Solutions/SIGNL4/Data Connectors/DerdackSIGNL4.json [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    "permissions": {
+        "resourceProvider": [
+            {
+                "provider": "Microsoft.OperationalInsights/workspaces",
+                "permissionsDisplayText": "read and write permissions are required.",
+                "providerDisplayName": "Workspace",
+                "scope": "Workspace",
+                "requiredPermissions": {
+                    "write": true,
+                    "read": true
```
Comment (NikTripathi) [inline Solutions/SIGNL4/Data Connectors/DerdackSIGNL4.json:46]: Please add delete permission also.

**Review by NikTripathi — CHANGES_REQUESTED** [review]

File: Solutions/SIGNL4/Data Connectors/Logo/signl4.svg (hunk 1 above)
Comment (rons4) [reply Solutions/SIGNL4/Data Connectors/Logo/signl4.svg]: Thank you for your review.
The SVG size is under 5kb now.
By the way, it seems I cannot open the link to the guidelines.

File: Solutions/SIGNL4/Data Connectors/Logo/signl4.svg (hunk 2 above)
Comment (rons4) [reply Solutions/SIGNL4/Data Connectors/Logo/signl4.svg]: OK and done. The id's are GUID's now.

File: Solutions/SIGNL4/Data Connectors/Logo/signl4.svg (hunk 3 above)
Comment (rons4) [reply Solutions/SIGNL4/Data Connectors/Logo/signl4.svg]: OK and done. All xmlns linke removed.

File: Solutions/SIGNL4/Data Connectors/DerdackSIGNL4.json (hunk 4 above)
Comment (rons4) [reply Solutions/SIGNL4/Data Connectors/DerdackSIGNL4.json]: OK and done. "Delete" permission added.

**Comment by anki-narravula** [comment]
Hi @rons4 - Thank you for creating playbook. Can you please make following changes to the playbook, it is not working in current state.

1. You can use this template for playbook  - https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json
2. As you see in template, pls create resources for Sentinel  and workflow
image
3. Make sure you configure managed identity for sentinel connection
4. User inputs need to go as deployment parameters like SIGNL4 details 
5. Once you make the changes, you can test by deploying using - https://portal.azure.com/#create/Microsoft.Template

**Comment by rons4** [comment]
Thanks @anki-narravula, I have created the PlayBook and exported it as template like this:
image

It seems to import again fine. However, I assume this is not the right way to do it.

So, now I have tried to follow your instructions and used the file you have mentioned as a start. I can import the template successfully. However, I am not sure if I am still missing something.

Thanks a lot in advance for your feedback and hits.

**Comment by anki-narravula** [comment]
> Thanks @anki-narravula, I have created the PlayBook and exported it as template like this: image
> 
> It seems to import again fine. However, I assume this is not the right way to do it.
> 
> So, now I have tried to follow your instructions and used the file you have mentioned as a start. I can import the template successfully. However, I am not sure if I am still missing something.
> 
> Thanks a lot in advance for your feedback and hits.

Yes you are doing right. Export as template is first step we do. Once we have template, we need to change few things as the same template (as it as) won't work for other users to deploy. Connections are not properly captures, many static configurations etc, parameters are not properly set. While changing the template, refer https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json as it has all best practices in it.

**Comment by rons4** [comment]
Thanks again @anki-narravula,

this is the original export
https://github.com/rons4/Azure-Sentinel/blob/master/Solutions/SIGNL4/Playbooks/azuredeploy_1.json

And this is the export with the adaptations from the template:
https://github.com/rons4/Azure-Sentinel/blob/master/Solutions/SIGNL4/Playbooks/azuredeploy_2.json

Both versions are imported fine in the tester.

I guess azuredeploy_2.json is close to what we need.

Do you have any further comments?

File: Solutions/SIGNL4/Data Connectors/DerdackSIGNL4.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        {
+            "name": "SIGNL4_CL",
+            "lastDataReceivedQuery": "SecurityIncident\n | where Labels contains \"SIGNL4\""
+        }
+    ],
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "SecurityIncident\n | where Labels contains \"SIGNL4\" | project IsConnected = LastLogReceived > ago(30d)"
```
[the remark this answers is not in the document]
Comment (rons4) [reply Solutions/SIGNL4/Data Connectors/DerdackSIGNL4.json]: Thanks @vmanojreddy, I changed the query and it should work now. SIGNL4 is not a typical data provider. It rather gets information from Azure Sentinel and then sends alerts to users or teams.

I hope all is fine now and please let me know if you have any further comments.

**Comment by anki-narravula** [comment]
> Thanks again @anki-narravula,
> 
> this is the original export https://github.com/rons4/Azure-Sentinel/blob/master/Solutions/SIGNL4/Playbooks/azuredeploy_1.json
> 
> And this is the export with the adaptations from the template: https://github.com/rons4/Azure-Sentinel/blob/master/Solutions/SIGNL4/Playbooks/azuredeploy_2.json
> 
> Both versions are imported fine in the tester.
> 
> I guess azuredeploy_2.json is close to what we need.
> 
> Do you have any further comments?

We still have hard coded subscription id, resource group etc. These values should be dynamic. I have created template for you, have a look and test it once. One thing was not clear to me, how do you want users to authenticate with signl4. Do you want to ask users go to edit connections --> and authorize after deployment or you want to capture the credentials during deployment (as parameters) and authenticate. 

[signl4_azuredeploy.zip](https://github.com/Azure/Azure-Sentinel/files/7627938/signl4_azuredeploy.zip)

**Comment by rons4** [comment]
Thanks @anki-narravula, I have tested your template and it seems to work fine.
Regarding the authentication (authorize after deployment or to capture the credentials during deployment as parameters) I am completely open. Whatever works best.
I have updated the Playbook and I hope all is fine now. Just let me know if you should have any further comments.

**Review by anki-narravula — APPROVED** [review]
@NikTripathi  @rons4 - Playbooks are approved.

**Comment by rons4** [comment]
@anki-narravula, Thanks a lot.

**Review by NikTripathi — APPROVED** [review]

**Comment by NikTripathi** [comment]
Can merge once validations pass.

**Comment by rons4** [comment]
Thanks a lot @NikTripathi. Is there anything I can do from my side to make the validation pass?
If so, can you please assist how to address the issues?

**Comment by NikTripathi** [comment]
> Thanks a lot @NikTripathi. Is there anything I can do from my side to make the validation pass? If so, can you please assist how to address the issues?

We are working on that. Will let you know if something needed. Thanks

**Comment by v-sabiraj** [comment]
Hey @rons4, We have resolved the issue for PR Validations failure and request you to please update your PR branch with the latest changes from **Azure:master** so that the PR Validations can be executed.... Thanks!!!

**Comment by rons4** [comment]
Thanks @v-sabiraj, I did what you have requested and not the checks seem to have passed successfully. I hope I did everything correctly and please let me know what I can do from my side as the next step.

**Comment by ashishsyal** [comment]
@rons4 - Can you share which checks are not passing ?..

**Comment by rons4** [comment]
Thanks @ashishsyal and sorry. I mean NOW all tests seem to have passed successfully (not NOT).
So from my side all seems to be fine.

**Comment by rons4** [comment]
Thank you all for your assistance. I will now proceed with packaging.

**Pull request merged**